addons/point_of_sale/models/pos_session.py

```diff
@@ -1595,190 +1643,216 @@ def _pos_ui_models_to_load(self):
             'res.users',
             'product.pricelist',
             'account.bank.statement',
-            'product.pricelist.item',
-            'product.category',
             'res.currency',
             'pos.category',
             'product.product',
             'product.packaging',
             'account.cash.rounding',
             'pos.payment.method',
             'account.fiscal.position',
-            'account.fiscal.position.tax',
         ]
-        if self.config_id.product_configurator:
-            models_to_load.extend(['product.attribute', 'product.attribute.value', 'product.template.attribute.value'])
         return models_to_load
 
     def _loader_params_res_company(self):
         return {
-            'domain': [('id', '=', self.company_id.id)],
-            'fields': [
-                'currency_id', 'email', 'website', 'company_registry', 'vat', 'name', 'phone', 'partner_id',
-                'country_id', 'state_id', 'tax_calculation_rounding_method'
-            ],
+            'search_params': {
+                'domain': [('id', '=', self.company_id.id)],
+                'fields': [
+                    'currency_id', 'email', 'website', 'company_registry', 'vat', 'name', 'phone', 'partner_id',
+                    'country_id', 'state_id', 'tax_calculation_rounding_method'
+                ],
+            }
         }
 
     def _get_pos_ui_res_company(self, params):
-        return self.env['res.company'].search_read(params['domain'], params['fields'])
+        company = self.env['res.company'].search_read(**params['search_params'])[0]
```

Comment:
do we sometimes give something else than "search_params"? Otherwise, we can directly set search_params parameters.